### PR TITLE
DGR-964: Bump Tested up to version to 6.6

### PR DESCRIPTION
**Pull request by kei178**
This PR updates the tested WP version up to 6.6. As commented, it also fixes the development and CI setup for PHPCS and corrects PHPCS violations.

I've also conducted manual testing from the UI to make sure everything works as expected.

Screenshot 2024-10-03 at 11 35 15

Screenshot 2024-10-03 at 11 35 51

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -32,22 +32,16 @@ jobs:
...
-            composer: 2
-            wordpress: '5.9'
+            wordpress: '6.5'
           - php: '7.4'
             composer: 2
-            wordpress: '5.9'
+            wordpress: '6.6'
+          - php: '8.2'
+            composer: 2
+            wordpress: '6.6'
```
Comment (kei178) [inline .github/workflows/ci.yml:44]: Updating the CI setup to test the current major WP version `6` with possible PHP versions.


c.f. 
- https://wordpress.org/documentation/article/supported-versions/
- https://make.wordpress.org/core/handbook/references/php-compatibility-and-wordpress-versions/

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -72,11 +66,14 @@ jobs:
...
+          git clone -b main https://github.com/WordPress/WordPress-Coding-Standards.git vendor/wp-coding-standards/wpcs
+          composer global config --no-plugins allow-plugins.dealerdirect/phpcodesniffer-composer-installer true
           composer global require wp-coding-standards/wpcs
           composer global require phpcompatibility/phpcompatibility-paragonie
           composer global require phpcompatibility/phpcompatibility-wp
-          sudo vendor/bin/phpcs --config-set installed_paths vendor/wp-coding-standards/wpcs,vendor/phpcompatibility/php-compatibility,vendor/phpcompatibility/phpcompatibility-paragonie,vendor/phpcompatibility/phpcompatibility-wp
+          composer global require phpcsstandards/phpcsutils
+          composer global require phpcsstandards/phpcsextra
+          sudo vendor/bin/phpcs --config-set installed_paths vendor/wp-coding-standards/wpcs,vendor/phpcompatibility/php-compatibility,vendor/phpcompatibility/phpcompatibility-paragonie,vendor/phpcompatibility/phpcompatibility-wp,vendor/phpcsstandards/phpcsutils,vendor/phpcsstandards/phpcsextra
```
Comment (kei178) [inline .github/workflows/ci.yml:76]: To fix the PHPCS setup for CI.

File: bin/init-wp-dev.sh [hunk 3]
```diff
@@ -8,13 +8,17 @@ vendor/bin/phpunit --version
 # Configure WordPress-Coding-Standards
 wpcs_dir="WordPress"
 wpcs_path="${PLUGIN_DIR}/vendor/squizlabs/php_codesniffer/Standards/${wpcs_dir}"
-git clone -b master https://github.com/WordPress/WordPress-Coding-Standards.git $wpcs_path
-vendor/bin/phpcs --config-set installed_paths $wpcs_path
+git clone -b main https://github.com/WordPress/WordPress-Coding-Standards.git $wpcs_path
+phpcsutils_path="${PLUGIN_DIR}/vendor/phpcsstandards/phpcsutils"
+phpcompatibility_path="${PLUGIN_DIR}/vendor/phpcompatibility/phpcompatibility-wp"
+phpcs_extra_path="${PLUGIN_DIR}/vendor/phpcsstandards/phpcsextra"
+vendor/bin/phpcs --config-set installed_paths $wpcs_path, $phpcsutils_path, $phpcompatibility_path, $phpcs_extra_path
```
Comment (kei178) [inline bin/init-wp-dev.sh:15]: To fix the PHPCS setup for development.

File: composer.json [hunk 4]
```diff
@@ -3,7 +3,9 @@
         "yoast/phpunit-polyfills": "^1.0",
         "squizlabs/php_codesniffer": "^3.3.1",
         "dealerdirect/phpcodesniffer-composer-installer": "^0.7",
-        "phpcompatibility/phpcompatibility-wp": "*"
+        "phpcompatibility/phpcompatibility-wp": "*",
+        "phpcsstandards/phpcsutils": "^1.0",
+        "phpcsstandards/phpcsextra": "^1.2.0"
```
Comment (kei178) [inline composer.json:8]: To fix the PHPCS setup for development.

File: includes/class-accredible-learndash-admin-menu.php [hunk 5]
```diff
@@ -23,7 +23,7 @@ public static function add() {
 			add_menu_page(
 				'Accredible LearnDash Add-on',
 				'Accredible LearnDash Add-on',
-				'administrator',
+				'manage_options',
```
Comment (kei178) [inline includes/class-accredible-learndash-admin-menu.php:26]: To fix the following PHPCS violation.

Screenshot 2024-10-03 at 11 03 29

We need to provide not a role but a capability for the 3rd argument of `add_menu_page` and `add_submenu_page`.

c.f.
- https://developer.wordpress.org/reference/functions/add_menu_page/
- https://wordpress.org/documentation/article/roles-and-capabilities/

File: includes/helpers/class-accredible-learndash-admin-form-helper.php [hunk 6]
```diff
@@ -21,7 +21,7 @@ class Accredible_Learndash_Admin_Form_Helper {
 		public static function value_attr( $obj, $key ) {
 			$obj = (array) $obj;
 			if ( ! empty( $obj ) && isset( $obj[ $key ] ) ) {
-				echo sprintf(
+				printf(
```
Comment (kei178) [inline includes/helpers/class-accredible-learndash-admin-form-helper.php:24]: Auto-correction by PHPCBF: [bbcfe99](https://github.com/accredible/accredible-learndash-add-on/pull/73/commits/bbcfe99c04c223b662cc1893025d0e0ac8630364)

File: includes/helpers/class-accredible-learndash-admin-table-helper.php [hunk 7]
```diff
@@ -93,7 +93,7 @@ public function build_table_rows( $table_data ) {
 				}
 				$row_cells .= '</tr>';
 
-				$index ++;
+				++$index;
```
Comment (kei178) [inline includes/helpers/class-accredible-learndash-admin-table-helper.php:96]: Auto-correction by PHPCBF: [bbcfe99](https://github.com/accredible/accredible-learndash-add-on/pull/73/commits/bbcfe99c04c223b662cc1893025d0e0ac8630364)

File: includes/models/class-accredible-learndash-model.php [hunk 8]
```diff
@@ -157,10 +157,8 @@ public static function validate( $data, $id = null ) {
 					if ( ! isset( $data[ $field ] ) || '' === trim( $data[ $field ] ) ) {
 						throw new Exception( esc_attr( $field ) . ' is a required field.' );
 					}
-				} else {
-					if ( array_key_exists( $field, $data ) && ( null === $data[ $field ] || '' === trim( $data[ $field ] ) ) ) {
+				} elseif ( array_key_exists( $field, $data ) && ( null === $data[ $field ] || '' === trim( $data[ $field ] ) ) ) {
```
Comment (kei178) [inline includes/models/class-accredible-learndash-model.php:160]: Auto-correction by PHPCBF: [bbcfe99](https://github.com/accredible/accredible-learndash-add-on/pull/73/commits/bbcfe99c04c223b662cc1893025d0e0ac8630364)

File: includes/helpers/class-accredible-learndash-admin-form-helper.php [hunk 9]
```diff
@@ -32,11 +32,11 @@ public static function value_attr( $obj, $key ) {
...
 		 *
 		 * @param mixed $value value.
-		 * @param mixed $default default value.
+		 * @param mixed $default_value default value.
 		 */
-		public static function html( $value, $default = 'N/A' ) {
+		public static function html( $value, $default_value = 'N/A' ) {
 			if ( empty( $value ) ) {
-				$value = $default;
+				$value = $default_value;
```
Comment (kei178) [inline includes/helpers/class-accredible-learndash-admin-form-helper.php:39]: To fix the following PHPCS warning:

Screenshot 2024-10-03 at 11 03 35

**Review by albertpaulp — APPROVED** [review]
My php foo is definitely rusted but changes look reasonable also CI & manual testing gives confidence, off to you go sailer 🚢

**Pull request merged**